tests/foreman/ui/test_gpgkey.py

```diff
@@ -393,10 +368,172 @@ def test_negative_update_2(self, new_name):
+        """@test: Hosts can install packages using gpg key associated with
+        multiple custom repositories
+
+        @feature: GPG Keys
+
+        @assert: host can install package from custom repositories
+
+        @status: manual
+
+        """
+
+        pass
+
+    @unittest.skip(NOT_IMPLEMENTED)
+    @data("""DATADRIVENGOESHERE
+        name is alpha
+        name is numeric
+        name is alphanumeric
+        name is utf-8
+        name is latin1
+        name is html
+        gpg key file is valid always
+""")
+    def test_consume_content_3(self):
+        """@test: Hosts can install packages using different gpg keys
+        associated with multiple custom repositories
+
+        @feature: GPG Keys
+
+        @assert: host can install package from custom repositories
+
+        @status: manual
+
+        """
+
+        pass
+
+    # Miscelaneous
+
+    @unittest.skip(NOT_IMPLEMENTED)
+    @data("""DATADRIVENGOESHERE
+        name is alpha
+        name is numeric
+        name is alphanumeric
+        name is utf-8
+        name is latin1
+        name is html
+        gpg key file is valid always
+""")
+    def test_list_key_1(self):
+        """@test: Create gpg key and list it
+
+        @feature: GPG Keys
+
+        @assert: gpg key is displayed/listed
+
+        @status: manual
+
+        """
+
+        pass
+
+    @unittest.skip(NOT_IMPLEMENTED)
+    @data("""DATADRIVENGOESHERE
+        name is alpha
+        name is numeric
+        name is alphanumeric
+        name is utf-8
+        name is latin1
+        name is html
+        gpg key file is valid always
+""")
+    def test_search_key_1(self):
+        """@test: Create gpg key and search/find it
+
+        @feature: GPG Keys
+
+        @assert: gpg key can be found
+
+        @status: manual
+
+        """
+
+        pass
+
+    @unittest.skip(NOT_IMPLEMENTED)
+    @data("""DATADRIVENGOESHERE
+        name is alpha
+        name is numeric
+        name is alphanumeric
+        name is utf-8
+        name is latin1
+        name is html
+        gpg key file is valid always
+""")
+    def test_info_key_1(self):
+        """@test: Create single gpg key and get its info
+
+        @feature: GPG Keys
+
+        @assert: specific information for gpg key matches the creation values
+
+        @status: manual
+
+        """
+
+        pass
+
+# TODO: Update the below tests with *generate_strings_list
+# After the below bug is fixed.
+# https://bugzilla.redhat.com/show_bug.cgi?id=1184480
+
 
-    @attr('ui', 'gpgkey', 'implemented')
-    @data(*generate_strings_list())
+@run_only_on('sat')
+@ddt
+class GPGKeyProductAssociate(UITestCase):
+    """Implements tests for GPG Keys via UI"""
+
+    @classmethod
+    def setUpClass(cls):
+        org_attrs = entities.Organization().create()
+        cls.org_name = org_attrs['name']
+        cls.org_id = org_attrs['id']
+
+        super(GPGKeyProductAssociate, cls).setUpClass()
+
+        # Product association
```

Comment:
Actually I had added a new class for `product association` and that is the result of copy and paste.